docs/proposals/20221115-static-pod-upgrade-model.md

```diff
@@ -0,0 +1,205 @@
+---
+title: Static Upgrade Model
+authors:
+  - ""
+reviewers:
+  - ""
+creation-date: 
+last-updated: 
+status: 
+---
+
+# Static Pod Upgrade Model
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Requirements (Optional)](#requirements-optional)
+      - [Functional Requirements](#functional-requirements)
+        - [FR1](#fr1)
+        - [FR2](#fr2)
+      - [Non-Functional Requirements](#non-functional-requirements)
+        - [NFR1](#nfr1)
+        - [NFR2](#nfr2)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+If this proposal adds new terms, or defines some, make the changes to the book's glossary when in PR stage.
+
+## Summary
+
+In this proposal, we hope to put forward a new method to control the upgrade of static pods automatically, so we add a new CRD(UpgradeableStaticPod) to control the upgrade of static pods. Meanwhile, we provide two upgrade strategy for static pod: Auto and OTA.
+
+## Motivation
+
+We all know that static pods are managed directly by the kubelet daemon on a specific node, without the API server observing them. Generally speaking, if we want to upgrade static pods, we may modify or replace the manifests under `staticPodPath`(Usually it's `/etc/kubernetes/manifests`) on a specific node. When the number of edge static pods becomes very large, this can be a truly tedious and painful task. Therefore, in cloud-edge collaboration scenario, it is very necessary to provide a proper static pod upgrade model like other workloads, such as DaemonSet.
+
+### Goals
+
+- Define the API of the new CRD
+- Provide corresponding controller
+- Support Auto and OTA upgrade strategy
+
```

Comment:
we also need to add Non Goals as following:
1. systemd service(like kubelet) upgrade will not be covered
2. do not consider the details of the static pod installation